portal-web/test/functional/com/liferay/portalweb/macros/RemoteApps.macro

```diff
@@ -88,28 +88,88 @@ definition {
 			locator1 = "TextInput#ANY",
 			value1 = "${entryName}");
 
+		if (isSet(entryDescription)) {
+			AssertTextEquals(
+				entryDescription = "${entryDescription}",
+				key_text = "_description",
+				locator1 = "TextArea#ANY",
+				value1 = "${entryDescription}");
+		}
+
+		if (isSet(entrySourceCodeURL)) {
+			AssertTextEquals(
+				entrySourceCodeURL = "${entrySourceCodeURL}",
+				key_text = " Source Code URL ",
+				locator1 = "TextInput#ANY_SPECIFIC",
+				value1 = "${entrySourceCodeURL}");
+		}
+
+		if (isSet(entryType)) {
+			AssertTextEquals(
+				entryType = "${entryType}",
+				key_id = "ClientExtensionAdminPortlet_typeLabel",
+				locator1 = "RemoteAppsEntry#URL",
+				value1 = "${entryType}");
+		}
+
+		if (isSet(entryCssurlText)) {
+			AssertTextEquals(
+				entryCssurlText = "${entryCssurlText}",
+				key_text = "CSS URL",
+				locator1 = "TextArea#ANY",
+				value1 = "${entryCssurlText}");
+		}
+
+		if (isSet(entryFriendlyElementName)) {
+			AssertTextEquals(
+				entryFriendlyElementName = "${entryFriendlyElementName}",
+				key_id = "ClientExtensionAdminPortlet_Friendly-URL-Mapping",
+				locator1 = "RemoteAppsEntry#URL",
+				value1 = "${entryFriendlyElementName}");
+		}
+
+		if (isSet(entryCssurl)) {
+			AssertTextEquals(
+				custom_entryCssurl = "${entryCssurl}",
+				key_text = "CSS URL",
+				locator1 = "TextInput#ANY",
+				value1 = "${entryCssurl}");
+		}
+
 		AssertTextEquals(
-			custom_entryHtmlName = "${entryHtmlName}",
+			entryHtmlName = "${entryHtmlName}",
 			key_text = "HTML Element Name",
 			locator1 = "TextInput#ANY",
 			value1 = "${entryHtmlName}");
 
-		AssertTextEquals(
-			custom_entryURL = "${entryURL}",
-			key_id = "ClientExtensionAdminPortlet_url",
-			locator1 = "RemoteAppsEntry#URL",
-			value1 = "${entryURL}");
+		if (isSet(entryCategoryName)) {
+			AssertTextEquals(
+				entryCategoryName = "${entryCategoryName}",
+				key_text = "Portlet Category Name",
+				locator1 = "TextInput#ANY",
+				value1 = "${entryCategoryName}");
+		}
 
-		AssertTextEquals(
-			custom_entryCssurl = "${entryCssurl}",
-			key_text = "CSS URL",
-			locator1 = "TextInput#ANY",
-			value1 = "${entryCssurl}");
+		if (isSet(entryURLText)) {
+			AssertTextEquals(
+				entryURLText = "${entryURLText}",
+				key_text = "ClientExtensionAdminPortlet_URLs",
+				locator1 = "TextArea#ANY",
+				value1 = "${entryURLText}");
+		}
+
+		if (isSet(entryURL)) {
+			AssertTextEquals(
+				custom_entryURL = "${entryURL}",
+				key_id = "ClientExtensionAdminPortlet_url",
+				locator1 = "RemoteAppsEntry#URL",
+				value1 = "${entryURL}");
+		}
 
 		AssertTextEquals(
-			custom_entryProperties = "${entryProperties}",
-			key_text = "Properties",
-			locator1 = "RemoteAppsEntry#REMOTE_APPS_PROPERTIES",
+			entryProperties = "${entryProperties}",
+			key_text = "properties",
+			locator1 = "TextArea#ANY",
 			value1 = "${entryProperties}");
```

Comment:
This macro looks extremely long, almost 100 lines... could we break this up into other macros?

Please see: https://liferay.slack.com/archives/GSMKNMV6Z/p1658944557952609

Reply:
or maybe create something like...assertAllFieldsDisabled and while loop through all TextInput/TextArea/etc. fields.

Reply:
I'm debating over this some more and the more I think about it, the more I agree with what you currently have. So don't change anything with the content of the two macros, this is probably best.